src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -574,27 +574,51 @@ public static FileAttributes GetAttributes(string fullPath)
             return attributes;
         }
 
+        public static FileAttributes GetAttributes(SafeFileHandle fileHandle)
+        => default(FileStatus).GetAttributes(fileHandle);
+
         public static void SetAttributes(string fullPath, FileAttributes attributes)
             => default(FileStatus).SetAttributes(fullPath, attributes, asDirectory: false);
 
+        public static void SetAttributes(SafeFileHandle fileHandle, FileAttributes attributes)
+        => default(FileStatus).SetAttributes(fileHandle, attributes, asDirectory: false);
+
         public static DateTimeOffset GetCreationTime(string fullPath)
             => default(FileStatus).GetCreationTime(fullPath).UtcDateTime;
 
+        public static DateTimeOffset GetCreationTime(SafeFileHandle fileHandle)
+        => default(FileStatus).GetCreationTime(fileHandle).UtcDateTime;
+
         public static void SetCreationTime(string fullPath, DateTimeOffset time, bool asDirectory)
             => default(FileStatus).SetCreationTime(fullPath, time, asDirectory);
 
+        public static void SetCreationTime(SafeFileHandle fileHandle, DateTimeOffset time)
+            => default(FileStatus).SetCreationTime(fileHandle, time, asDirectory: false);
+
         public static DateTimeOffset GetLastAccessTime(string fullPath)
             => default(FileStatus).GetLastAccessTime(fullPath).UtcDateTime;
 
+        public static DateTimeOffset GetLastAccessTime(SafeFileHandle fileHandle)
+            => default(FileStatus).GetLastAccessTime(fileHandle).UtcDateTime;
+
         public static void SetLastAccessTime(string fullPath, DateTimeOffset time, bool asDirectory)
             => default(FileStatus).SetLastAccessTime(fullPath, time, asDirectory);
 
+        public static unsafe void SetLastAccessTime(SafeFileHandle fileHandle, DateTimeOffset time)
+            => default(FileStatus).SetLastAccessTime(fileHandle, time, asDirectory: false);
+
         public static DateTimeOffset GetLastWriteTime(string fullPath)
             => default(FileStatus).GetLastWriteTime(fullPath).UtcDateTime;
 
+        public static DateTimeOffset GetLastWriteTime(SafeFileHandle fileHandle)
+            => default(FileStatus).GetLastWriteTime(fileHandle).UtcDateTime;
+
         public static void SetLastWriteTime(string fullPath, DateTimeOffset time, bool asDirectory)
             => default(FileStatus).SetLastWriteTime(fullPath, time, asDirectory);
 
+        public static unsafe void SetLastWriteTime(SafeFileHandle fileHandle, DateTimeOffset time)
+            => default(FileStatus).SetLastWriteTime(fileHandle, time, asDirectory: false);
```

Comment:
Why are these called methods instance methods rather than statics?

Reply:
I think this is due to the design of the `FileStatus` struct. There are hardly any static methods for `FileStatus`.

Reply:
`FileStatus` has instance methods because it was designed for `FileSystemInfo`.

For previous .NET versions, this was also implemented using instance methods, which created a FSI:
```cs
        public static void SetLastWriteTime(string fullPath, DateTimeOffset time, bool asDirectory)
        {
            FileSystemInfo info = asDirectory ?
                (FileSystemInfo)new DirectoryInfo(fullPath, null) :
                (FileSystemInfo)new FileInfo(fullPath, null);

            info.LastWriteTimeCore = time;
        }
```